.github/workflows/ubuntu.yml

```diff
@@ -2,9 +2,6 @@ name: ubuntu
 
 on: [push, pull_request]
 
-env:
```

Comment:
I think this and next commit where you add `-D werror=true` need to be squashed. Otherwise current commit which drops CFLAGS basically regress experience.

Unfortunately combination of this and next commit still causes regression in experience for autotools build since you drop `-Werror` and don't add it back.

Also, I am highly unsure that `-Wall` is getting added by default in both autools/meson scripts. As a result you basically reduce set of warnings we monitor.

Reply:
Sure can squash.

Don't fully understand your concern with "regression" - all the code that's being build with autoconf is already build with meson - on even more platforms.

Does that make sense or am I simply too tired and misread the code?

Reply:
It's also safe to just reverse the order... If -Werror is passed twice, once by meson and once by CFLAGS, that's redundant but harmless and the follow-up commit can be marketed as a mere cleanup.

...

Meson definitely defaults to -Wall, unless you specify a non-default `warning_level=0` option.

Reply:
@dvrogozh for meson we use `warning_level=1`  which gives us `-Wall` plus a bit more. On the configure side - _no_ compiler options are passed at all. Even so, as I said earlier meson already compiles the autoconf covered code so we should be all good.

Happy to make changes though - just clearly put your suggestion.

Reply:
This PR started as a fix for Windows warning and Windows CI where we missed `-Werror`. I suggest to fix exactly what is being missed - Windows part plus FreeBSD part where we also miss the flag. If you want to use meson way of fixing that - I am all fine.

However, I am not ok with the change to Linux part of the ci where we already had `-Werror` and where you are dropping it for autotools part. I do not believe that's a right thing to do. Autotools build path is still a way to go for a number of people and I do want this to get tested. I am not aligned with the motivation that we already test this with meson. Maybe or maybe not - it's much easier to just test autotools with `Werror` and be sure w/o any time spent on arguing. Same goes with `-Wall` - I believe it's reasonable to get build tested with all the warnings enabled. Current way of testing explicitly enables that on CI level. Just leave that as is.

Summarizing Linux part - I would prefer to leave it as-is without any modifications and just do Windows & freebsd.

Reply:
Thanks for elaborating. Your earlier suggestion "we can have one but not both" was pointing at another direction.

Kept the `CFLAGS` piece and dropped the meson `-D error=true` from the Linux/Ubuntu build. Hope that's better